src/Native/Runtime/arm/PInvoke.S

```diff
@@ -3,39 +3,77 @@
 // See the LICENSE file in the project root for more information.
 
 #include <unixasmmacros.inc>
+#include <AsmOffsets.inc>
 
 .syntax unified
 .thumb
 
-#ifdef _DEBUG
-.rodata
-AssertMsg:                              .asciz "__FILE__:%s: %s is not implemented\n"
-FileName:                               .asciz "PInvoke.S"
-RhpPInvokeName:                         .asciz "RhpPInvoke"
-RhpPInvokeReturnName:                   .asciz "RhpPInvokeReturn"
+.set __tls_array, 0x2C    // offsetof(TEB, ThreadLocalStoragePointer)
 
-.text
-.macro GEN_ASSERT_FUNC func
-        GEN_ASSERT AssertMsg, FileName, \func
-.endm
-#endif
+.global __SECTIONREL_tls_CurrentThread
+__SECTIONREL_tls_CurrentThread:  .string "SECTIONREL_tls_CurrentThread"
 
-// TODO: Implement Arm support
+.macro 	INLINE_GETTHREAD destReg,trashReg
+.global _tls_index
+	ldr         \destReg, =_tls_index
```

Comment:
This looks like a direct translation of the Windows code. The Unix ABI for thread local access is quite different. This code needs to be written against the Unix ABI.